gems/chunky_png/1.4.0/chunky_png.rbs

```diff
@@ -0,0 +1,1362 @@
+    def <=>: (instance other) -> Integer
+
+    # Casts this dimension into an array.
+    # @return [Array<Integer>] <tt>[width, height]</tt> for this dimension.
+    def to_a: () -> ::Array[Integer]
+
+    alias to_ary to_a
+
+    # Initializes a new dimension instance.
+    # @param [Integer] width The width-component of the new dimension.
+    # @param [Integer] height The height-component of the new dimension.
+    def initialize: (Integer width, Integer height) -> instance
+  end
+
+  # Class that represents a vector of points, i.e. a list of {ChunkyPNG::Point} instances.
+  #
+  # Vectors can be created quite flexibly. See the {ChunkyPNG.Vector} factory methods for
+  # more information on how to construct vectors.
+  class Vector
+    include Enumerable[untyped]
+
+    # @return [Array<ChunkyPNG::Point>] The array that holds all the points in this vector.
+    attr_reader points: Array[ChunkyPNG::Point]
+
+    # Initializes a vector based on a list of Point instances.
+    #
+    # You usually do not want to use this method directly, but call {ChunkyPNG.Vector} instead.
+    #
+    # @param [Array<ChunkyPNG::Point>] points
+    # @see ChunkyPNG.Vector
+    def initialize: (?Array[ChunkyPNG::Point] points) -> instance
+
+    # Iterates over all the edges in this vector.
+    #
+    # An edge is a combination of two subsequent points in the vector. Together, they will form
+    # a path from the first point to the last point
+    #
+    # @param [true, false] close Whether to close the path, i.e. return an edge that connects the last
+    #   point in the vector back to the first point.
+    # @return [void]
+    # @raise [ChunkyPNG::ExpectationFailed] if the vector contains less than two points.
+    # @see #edges
+    def each_edge: (?bool close) { (untyped, untyped) -> void } -> void
+
+    # Returns the point with the given indexof this vector.
+    # @param [Integer] index The 0-based index of the point in this vector.
+    # @return [ChunkyPNG::Point] The point instance.
+    def []: (Integer index) -> ChunkyPNG::Point
+
+    # Returns an enumerator that will iterate over all the edges in this vector.
+    # @param (see #each_edge)
+    # @return [Enumerator] The enumerator that iterates over the edges.
+    # @raise [ChunkyPNG::ExpectationFailed] if the vector contains less than two points.
+    # @see #each_edge
+    def edges: (?bool close) -> Enumerator[untyped, void]
+
+    # Returns the number of points in this vector.
+    # @return [Integer] The length of the points array.
+    def length: () -> Integer
+
+    # Iterates over all the points in this vector
+    # @yield [ChunkyPNG::Point] The points in the correct order.
+    # @return [void]
+    def each: () { () -> untyped } -> void
+
+    # Comparison between two vectors for quality.
+    # @param [ChunkyPNG::Vector] other The vector to compare with.
+    # @return [true, false] true if the list of points are identical
+    def eql?: (instance other) -> bool
```

Comment:
I usually use `untyped` for the argument of equality operator, but the specific one is correct in this case. The implementation doesn't check the class of `other` value.

Reply:
Sir, I update the type of the parameter from `instance` to `untyped`, hence resolving the conversation.